Assets/Scripts/Entities/Player.cs

```diff
@@ -30,6 +38,41 @@ protected override void Awake()
             spriteRenderer.color = Random.ColorHSV();
         }
 
+        public void Update()
+        {
+            List<Collider2D> results = new List<Collider2D>();
+            ContactFilter2D filter = new ContactFilter2D
+            {
+                layerMask = buildingLayer,
+                useLayerMask = true
+            };
+            // Check if building in range
+            if (Physics2D.OverlapCircle(transform.position, upgradeMenuToggleRange, filter, results) == 0)
```

Comment:
Kannst das in FixedUpdate packen? Alles was Physics-Related ist, sollte da rein

Reply:
Done 👍